utils/conditions/conditions.go

```diff
@@ -64,10 +64,13 @@ const (
 	// RolloutUpdatedMessage indicates the rollout spec was updated.
 	RolloutUpdatedMessage = "Rollout updated to revision %s"
 
+	// RolloutHealthyReason is added in a rollout when it is healthy.
+	RolloutHealthyReason = "RolloutHealthy"
+	// RolloutHealthyMessage is added when the rollout is healthy
+	RolloutHealthyMessage = "Rollout healthy update to revision %d (%s): %s"
+
```

Comment:
Including revision number doesn't make sense anymore for when a Rollout becomes Healthy. I feel we need to emit two events, one for each condition (Healthy vs. Completed). Users will care separately when a Rollout finishes an update, but they will also care about from when a Rollout degrades (e.g. because of a pod restart) and subsequently recovers.

Reply:
The only place this is used is here https://github.com/argoproj/argo-rollouts/pull/2147/files/ee78ef0b85d3af3b21cd533930dc10b35b70c78d#diff-594206431376a15241cfdd95572e5d6e586929f867ca28ff63d1f679c0334d77R936-R938 which I added a TODO as well because I did not think it was the correct event to emit. I look at removing that message and changing that event to something that makes sense.

Reply:
Ah yes. I think that part of the code should not change and we should continue to emit `RolloutCompletedReason` when we switch stable hash to current hash. Not that the rollout is Healthy.